naga/src/back/hlsl/writer.rs

```diff
@@ -2719,16 +2749,40 @@ impl<'a, W: fmt::Write> super::Writer<'a, W> {
                 write!(self.out, ")")?;
             }
 
+            Expression::Binary {
+                op: crate::BinaryOperator::Divide,
+                left,
+                right,
+            } if matches!(
+                func_ctx.resolve_type(expr, &module.types).scalar_kind(),
+                Some(ScalarKind::Sint | ScalarKind::Uint)
+            ) =>
+            {
+                write!(self.out, "{DIV_FUNCTION}(")?;
+                self.write_expr(module, left, func_ctx)?;
+                write!(self.out, ", ")?;
+                self.write_expr(module, right, func_ctx)?;
+                write!(self.out, ")")?;
+            }
+
+            Expression::Binary {
+                op: crate::BinaryOperator::Modulo,
+                left,
+                right,
+            } if matches!(
+                func_ctx.resolve_type(expr, &module.types).scalar_kind(),
+                Some(ScalarKind::Sint | ScalarKind::Uint)
+            ) =>
+            {
+                write!(self.out, "{MOD_FUNCTION}(")?;
+                self.write_expr(module, left, func_ctx)?;
+                write!(self.out, ", ")?;
+                self.write_expr(module, right, func_ctx)?;
+                write!(self.out, ")")?;
+            }
+
             // TODO: handle undefined behavior of BinaryOperator::Modulo
```

Comment:
Can we delete this comment too?

Reply:
Where does HLSL say floating-point division by zero is UB? It seems like they usually follow IEEE.

Reply:
This table is describing DXIL, and seems to imply that it's defined.

https://github.com/microsoft/DirectXShaderCompiler/blob/main/docs/DXIL.rst#fdiv

Reply:
More links: https://microsoft.github.io/DirectX-Specs/d3d/archive/D3D11_3_FunctionalSpec.htm#3.1%20Floating%20Point%20Rules

Reply:
Sorry I meant to say modulo not division. But you might be right we can remove the comment, or at least have to update it. The modulus operator says it's undefined unless both operands are positive or negative. But we don't use the modulus operator, we use `fmod()`. This doesn't say anything about being undefined. But we do need to check whether it has the same semantics as WGSL..

Reply:
Removed the reference to undefined behaviour, but left the "TODO: if right == 0 return ?" part for now.